core/animated-entity.js

```diff
@@ -0,0 +1,58 @@
+import { create_float_buffer, create_index_buffer } from './context';
+import { Entity } from './entity';
+
+export class AnimatedEntity extends Entity {
+  constructor(options) {
+    super(options);
+
+    if (!this.frames || !this.frames.length) {
+      this.buffers = [];
+    }
+
+    this.current_frame = 0;
+    this.next_frame = 1;
+    this.current_tick = 0;
```

Comment:
It would help a lot if these properties had one-line comments associated with them.